core/src/main/java/hudson/PluginManager.java

```diff
@@ -506,6 +508,15 @@ protected boolean filter(Method e) {
         }
 
         LOGGER.info("Plugin " + p.getShortName()+":"+p.getVersion() + " dynamically installed");
+
+        // If this plugin is a UIThemeContributor, tell the UIThemesProcessor about it.
+        // TODO: I don't see a plugin uninstall sequence ?
+        if (isUIThemesPluginInstalled()) {
+            Plugin pluginInstance = p.getPlugin();
+            if (pluginInstance instanceof UIThemeContributor) {
```

Comment:
I agree with [the comment in the mailing list](https://groups.google.com/forum/?hl=en#!topic/jenkinsci-dev/VaTOWEZHyfI) that `UIThemeContributor` should be an extension point. That's how every other extension point is designed. For the common idiom of defiing an extension point, see `PingFailureAnalyzer` in the core. As for what happens when we get to support dynamic plugin uninstallation, if we ever do that I expect `ExtensionList` to automatically reflect the changes.

I generally recommend extension points like `UIThemeContributor` be abstract class to allow for future evolutions with compatibility, but if it's too late for that, that's OK.

This change would eliminate the need for this hunk. If it'd be quicker for us to be on Hangout to get this change done in this PR and the uithemes-processor plugin, let me know and I'd be happy to do that.